### PR TITLE
Verification restart after timeout

### PR DESCRIPTION
**Pull request by Elfoniok**

### CODE REVIEW

File: apps/blender/verification_queue.py [hunk 1]
```diff
@@ -3,23 +3,28 @@
...
 class VerificationQueue:
 
+    VERIFICATION_TIMEOUT = 90
+    RESTART_DELAY = 5
+
     def __init__(self, concurrency: int = 1) -> None:
         self._concurrency = concurrency
         self._queue: queue.Queue = queue.Queue()
         self._jobs: Dict[str, Deferred] = dict()
+        self.tasks: Dict[VerificationTask, Deferred] = dict()
```
Comment (mfranciszkiewicz) [inline apps/blender/verification_queue.py]: `self.tasks` seems redundant - it's only used for writing.

File: apps/blender/verification_queue.py [hunk 2]
```diff
@@ -3,23 +3,28 @@
...
         self.callbacks: Dict[VerificationTask, FunctionType] = dict()
         self._paused = False
+        self.timed_out = False
```
Comment (mfranciszkiewicz) [inline apps/blender/verification_queue.py]: Maybe `self._timed_out`?

File: apps/blender/verification_queue.py [hunk 3]
```diff
@@ -79,12 +84,37 @@ def callback(*args):
...
+            fn_timeout = partial(self.stop_task, task=entry, event=result,
+                                 subtask_id=subtask_id,
+                                 verifier_cls=verifier_cls)
+
+            result.addTimeout(VerificationQueue.VERIFICATION_TIMEOUT, reactor,
+                              onTimeoutCancel=fn_timeout)
             self._jobs[subtask_id] = result
+            self.tasks[entry] = result
+
+    def stop_task(self, *_, **kwargs):
```
Comment (mfranciszkiewicz) [inline apps/blender/verification_queue.py]: `stop_task` is not the same as "timed out". I think it's worth renaming and hiding (e.g. as `_verification_timed_out`). 

Plus, maybe you can use named keyword arguments (`task`, `event`, etc.) for readability?

File: apps/blender/verification_queue.py [hunk 4]
```diff
@@ -79,12 +84,37 @@ def callback(*args):
                 self._jobs.pop(subtask_id, None)
                 self._process_queue()
 
+        def errback(_):
+            if self.timed_out:
+                logger.debug("Timeout detected")
```
Comment (mfranciszkiewicz) [inline apps/blender/verification_queue.py]: A warning / error perhaps? E.g. "Verification timed out for subtask %r"

File: apps/blender/verification_queue.py [hunk 5]
```diff
@@ -79,12 +84,37 @@ def callback(*args):
...
+                reactor.callLater(VerificationQueue.RESTART_DELAY,
+                                  self._process_queue)
+                self.timed_out = False
+            else:
+                logger.debug("Finishing verification with fail")
```
Comment (mfranciszkiewicz) [inline apps/blender/verification_queue.py]: I'm not sure if the message will be duplicated, but this one could use a higher log severity.

File: apps/blender/verification_queue.py (hunk 1 above)
Comment (Elfoniok) [reply apps/blender/verification_queue.py]: done

File: apps/blender/verification_queue.py (hunk 2 above)
Comment (Elfoniok) [reply apps/blender/verification_queue.py]: done

File: apps/blender/verification_queue.py (hunk 3 above)
Comment (Elfoniok) [reply apps/blender/verification_queue.py]: done

File: apps/blender/verification_queue.py (hunk 4 above)
Comment (Elfoniok) [reply apps/blender/verification_queue.py]: done

File: apps/blender/verification_queue.py (hunk 5 above)
Comment (Elfoniok) [reply apps/blender/verification_queue.py]: done

File: apps/blender/verification_task.py [hunk 6]
```diff
@@ -1,30 +1,34 @@
...
         else:
-            verifier.task_timeout(self.subtask_id)
+            self.verifier.task_timeout(self.subtask_id)
         return None
 
+    def get_results(self):
+        return self.verifier.verification_completed()
+
     # Currently golem protocol does not allow for partial verification,
     # therefore we have to wait for any ongoing verification.
```
Comment (jiivan) [inline apps/blender/verification_task.py:31]: Is this comment still valid?

**Review by jiivan — APPROVED** [review]
LGTM

File: apps/blender/verification_task.py (hunk 6 above)
Comment (Elfoniok) [reply apps/blender/verification_task.py:31]: That is a good question, currently, verification task can be restarted after canceling, so this means that is not entirely valid. But I will remove it when proper verification queue persistence is implemented.

**Pull request merged**